datumaro/plugins/synthetic_data/image_generator.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from multiprocessing import Pool
+from random import Random
+from shutil import rmtree
+from typing import List, Optional, Tuple
+import logging as log
+import os
+import os.path as osp
+
+import cv2 as cv
+import numpy as np
+
+from datumaro.cli.util.errors import CliException
+from datumaro.components.cli_plugin import CliPlugin
+
+from .utils import IFSFunction, augment, colorize, download_colorization_model
+
+
+class ImageGenerator(CliPlugin):
+    """
+    ImageGenerator generates a 3-channel synthetic images with provided shape.
+    """
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument('-o', '--output-dir', type=osp.abspath, required=True,
+            help="Path to the directory where dataset are saved")
+        parser.add_argument('-k', '--count', type=int, required=True,
+            help="Number of images to generate")
+        parser.add_argument('--shape', nargs='+', type=int, required=True,
+            help="Image shape: height and width, for example --shape 256 224")
+        parser.add_argument('--overwrite', action='store_true',
+            help="Overwrite existing files in the save directory")
+
+        return parser
+
+    def __init__(self, output_dir: str, count: int,
+                 shape: Tuple[int, int], overwrite: bool = False):
+        self._count = count
+        self._output_dir = output_dir
+        assert len(shape) == 2
+        self._height, self._width = shape
+
+        if not osp.exists(output_dir):
+            os.mkdir(output_dir)
+        elif osp.isdir(output_dir) and os.listdir(output_dir):
+            if overwrite:
+                rmtree(output_dir)
+                os.mkdir(output_dir)
+            else:
+                raise CliException("Directory '%s' already exists "
+                    "(pass --overwrite to overwrite)" % output_dir)
+
+        self._rng = Random(0)
+        self._cpu_count = min(os.cpu_count(), self._count)
+        self._weights = np.array([
+            0.2, 1, 1, 1, 1, 1,
+            0.6, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1,
+            1.4, 1, 1, 1, 1, 1,
+            1.8, 1, 1, 1, 1, 1,
+            1, 0.2, 1, 1, 1, 1,
+            1, 0.6, 1, 1, 1, 1,
+            1, 1.4, 1, 1, 1, 1,
+            1, 1.8, 1, 1, 1, 1,
+            1, 1, 0.2, 1, 1, 1,
+            1, 1, 0.6, 1, 1, 1,
+            1, 1, 1.4, 1, 1, 1,
+            1, 1, 1.8, 1, 1, 1,
+            1, 1, 1, 0.2, 1, 1,
+            1, 1, 1, 0.6, 1, 1,
+            1, 1, 1, 1.4, 1, 1,
+            1, 1, 1, 1.8, 1, 1,
+            1, 1, 1, 1, 0.2, 1,
+            1, 1, 1, 1, 0.6, 1,
+            1, 1, 1, 1, 1.4, 1,
+            1, 1, 1, 1, 1.8, 1,
+            1, 1, 1, 1, 1, 0.2,
+            1, 1, 1, 1, 1, 0.6,
+            1, 1, 1, 1, 1, 1.4,
+            1, 1, 1, 1, 1, 1.8,
+        ]).reshape(-1, 6)
```

Comment:
If it okay that only the 3-rd line has all 1-s? 
What does 0.4 mean?
Can we generate it with a more clean formulae?

Reply:
weights from original paper https://openaccess.thecvf.com/content/ACCV2020/papers/Kataoka_Pre-training_without_Natural_Images_ACCV_2020_paper.pdf

Reply:
Please add a reference in comments

Reply:
Having studied the article, it seems to me that the current way of representing the weights obscures the principle by which they are generated. I would suggest to replace it with something like the following:

```python
BASE_WEIGHTS = np.ones((IfsFunction.NUM_PARAMS,))
WEIGHT_INTERVAL = 0.4
INTERVAL_MULTIPLIERS = (-2, -1, 1, 2)
weight_vectors = [BASE_WEIGHTS]

for weight_index in range(IfsFunction.NUM_PARAMS):
    for multiplier in INTERVAL_MULTIPLIERS:
        modified_weights = BASE_WEIGHTS.copy()
        modified_weights[weight_index] += multiplier * WEIGHT_INTERVAL
        weight_vectors.append(modified_weights)

# convert `weight_vectors` to a NumPy array if needed
```

`NUM_PARAMS` should be added as a constant to `IfsFunction`, since that's where this number logically stems from.

Reply:
A method has been created to fill in the weights